modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/integration/OpenApiServlet.java

```diff
@@ -80,4 +106,43 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         }
 
     }
+
+    public static Map<String, List<String>> getQueryParams(String queryString) {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        if (StringUtils.isEmpty(queryString)) {
+            return queryParameters;
+        }
+
+        String[] parameters = queryString.split("&");
+
+        for (String parameter : parameters) {
+            String[] keyValuePair = parameter.split("=");
+            queryParameters.put(keyValuePair[0], Collections.singletonList(keyValuePair[1]));
+        }
+        return queryParameters;
+    }
+
+    private static Map<String, String> getCookies(Cookie[] cookies) {
+        Map<String, String> mapOfCookies = new HashMap<>();
+        if (cookies != null) {
+            for (Cookie cookie : cookies) {
+                mapOfCookies.put(cookie.getName(), cookie.getValue());
+            }
+        }
+        return mapOfCookies;
```

Comment:
There can be multiple cookies with the same name but we would only return the last cookie for any name in the map.

Consider using [`MultivaluedHashMap`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MultivaluedHashMap.html) instead.

Reply:
I agree, but the cookie parameter for io.swagger.v3.core.filter.SpecFilter#filter() and the underlying abstract and interface (OpenAPISpecFilter) only takes in an Map<String, String>, so a MultivaluedHashMap won't work. Skimming through these classes, it seems the cookies aren't used for anything yet, I can't find an filterCookies() in SpecFilter so that may be the reason it is only taking in an Map<String, String>, instead of and Map<String, List<String>>. Do you want be to refactor the whole thing, or just leave it as it is. Could create another PR for this refactoring covering both OpenApiServlet and BaseOpenApiResource.

Reply:
Fair enough, let's keep the "flat" map for now.

@espenaf Could you please create an issue for refactoring the Cookie handling accordingly? 😃

Reply:
Issue created in https://github.com/swagger-api/swagger-core/issues/3582